src/main/native/GCM.c

```diff
@@ -136,504 +128,335 @@ ICC_AES_GCM_CTX* getOrfreeGCMContext(ICC_CTX* ockCtx, int keyLen) {
+					}
+#endif					
+					
 					return ICC_AES_GCM_CRYPTUPDATE_FAILED;
 				}
-			} else {
-				ockCheckStatus(ockCtx);
-				return ICC_AES_GCM_CRYPTUPDATE_FAILED;
-			}
+				else {
+					
+#ifdef DEBUG_GCM_DETAIL
+	   				if ( debug ) {
+    	    			gslogMessage ("DETAIL_GCM NI AAD ICC_AES_GCM_DecryptUpdate Succeeded\n");
+    				}
+#endif
+				}
+             }
+             else {
+             	
+#ifdef DEBUG_GCM_DETAIL
+	   			if ( debug ) {
+    	    		gslogMessage ("DETAIL_GCM NI No AAD ICC_AES_GCM_DecryptUpdate not called \n");
+    			}
+#endif
+             	
+             }
+        
 		} else {
 			ockCheckStatus(ockCtx);
+#ifdef DEBUG_GCM_DETAIL
+	   		if ( debug ) {
+    	    	gslogMessage ("DETAIL_GCM NI ICC_AES_GCM_DecryptUpdate ICC_AES_GCM_INIT_FAILED \n");
+    		}
+#endif
+			
 			return ICC_AES_GCM_INIT_FAILED;
 		}
 	} else {
 		/* GCM CTX Failed - no need to free it. */
 		ockCheckStatus(ockCtx);
+		
+#ifdef DEBUG_GCM_DETAIL
+	   		if ( debug ) {
+    	    	            gslogMessage ("DETAIL_GCM NI gslogMessage ICC_AES_GCM_NEW_FAILED\n" );
+    		}
+#endif
+		
 		return ICC_AES_GCM_CTX_NEW_FAILED;
 	}
 	return 0;
+	
+#else
+      	return -1;
+#endif
 }
+/*============================================================================
+*
+* Class:     com_ibm_crypto_plus_provider_ock_NativeInterface
+* Method:    internal functions
+* Signature: 
+*/
+int GCM_UpdForUpdateDecrypt_core (JNIEnv *env, ICC_CTX* ockCtx, ICC_AES_GCM_CTX* gcmCtx,
+	unsigned char * data, int dataOffset, int dataLen,
+	unsigned char * out, int outOffset,
+	unsigned long *updateOutlen) {
+#if defined(AIX) || defined(WINDOWS) || defined(MAC) || defined (LINUX) || defined(__MVS__)
 
-
-
-/*
- *  Class:     com_ibm_crypto_plus_provider_ock_NativeInterface
- *  Method:    do_GCM_decryptFastJNI_WithHardwareSupport
- *  */
-JNIEXPORT jint JNICALL Java_com_ibm_crypto_plus_provider_ock_NativeInterface_do_1GCM_1decryptFastJNI_1WithHardwareSupport
-(JNIEnv *env, jclass unusedClass, jint keyLen, jint ivLen, jint ciphertextOffset, jint ciphertextLen, jint plaintextOffset, 
-	jint aadLen, jint tagLen, jlong parameterBuffer, jbyteArray inputJ, jint inputOffset, jbyteArray outputJ, jint outputOffset) {
+	/* long unsigned updateAADlen   = 0; */
+	/* long unsigned finalOutlen    = 0;*/
+	int           rc             = ICC_OSSL_SUCCESS;
 	
-	// Setting static values
-	int            J0Offset = 64;
-	int            blockSize = 16;
-	int            counterValueOffset = 12;
-	int            tagOffset = 16;
-	int            keyOffset = 80;
-	long           mode = 0;
-	long           len = 0;
-	long           alen = 0;
-	jboolean       isCopy = 0;
-	unsigned char* input = NULL;
-	unsigned char* output = NULL;
-	unsigned char* parameters = NULL;
-	unsigned char* iv = NULL;
-	unsigned char* aad = NULL;
-	unsigned char* parm_block = NULL;
+	static const char * functionName = "NativeInterface.GCM_UpdForUpdateDecrypt_core";
```

Comment:
Same as before. Should we leave a space after the `*` too?

Reply:
I think we should have no space after the * only before them.